<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<link rel="stylesheet" type="text/css" href="../slides.css">
  <title>Horstmann Chapter 15</title>
</head>

<body>
<div onclick="document.location='slide021.html'">
<center><small><a href="slide019.html">previous</a> | 
<a href="slide001.html">start</a> | 
<a href="slide021.html">next</a></small></center>
<hr></div>
<h2>Answers</h2>
<ol>
<li value="5">The <tt>FileReader</tt> constructor succeeds, and in is constructed. Then the call
<tt>in.next()</tt> throws a <tt>NoSuchElementException</tt>, and the <tt>try</tt> block is aborted. None
of the <tt>catch</tt> clauses match, so none are executed. If none of the enclosing
method calls catch the exception, the program terminates.</li>
<li>No&#150;you catch both exception types in the same way, as you can see from the
code example on page 558. Recall that <tt>IOException</tt> is a checked exception and
<tt>NumberFormatException</tt> is an unchecked exception.</li>
</ol>
<br>

 

<div onclick="document.location='slide021.html'"><hr>
<center><small><a href="slide019.html">previous</a> | 
<a href="slide001.html">start</a> | 
<a href="slide021.html">next</a></small></center>
</div>
</body>
</html>
